test/common/config/utility_test.cc

```diff
@@ -293,6 +293,155 @@ TEST(UtilityTest, PrepareDnsRefreshStrategy) {
+
+      BackOffStrategyPtr strategy =
+          Utility::prepareBackoffStrategy<envoy::config::core::v3::GrpcService>(
+              api_config_source.grpc_services()[0], random);
+      EXPECT_NE(nullptr, dynamic_cast<JitteredExponentialBackOffStrategy*>(strategy.get()));
+
+      EXPECT_EQ(false, dynamic_cast<JitteredExponentialBackOffStrategy*>(strategy.get())
+                           ->isOverTimeLimit(10000));
+
+      EXPECT_EQ(true, dynamic_cast<JitteredExponentialBackOffStrategy*>(strategy.get())
+                          ->isOverTimeLimit(10001));
+    }
+  }
+}
+
+TEST(UtilityTest, prepareBackoffStrategyCustomValues) {
+  NiceMock<Random::MockRandomGenerator> random;
+  {
+    // set custom values for both base and max interval
+    {
+      envoy::config::core::v3::GrpcService grpc_service;
+
+      uint64_t test_base_interval_ms = 5000;
+      uint64_t test_max_interval_ms = 20000;
+
+      grpc_service.mutable_retry_policy()
+          ->mutable_retry_back_off()
+          ->mutable_base_interval()
+          ->set_seconds(test_base_interval_ms / 1000);
+      grpc_service.mutable_retry_policy()
+          ->mutable_retry_back_off()
+          ->mutable_max_interval()
+          ->set_seconds(test_max_interval_ms / 1000);
+
+      BackOffStrategyPtr strategy =
+          Utility::prepareBackoffStrategy<envoy::config::core::v3::GrpcService>(grpc_service,
+                                                                                random);
+
+      // provided time limit is equal to max time limit
+      EXPECT_EQ(false, dynamic_cast<JitteredExponentialBackOffStrategy*>(strategy.get())
+                           ->isOverTimeLimit(test_max_interval_ms));
+
+      // provided time limit is over max time limit
+      EXPECT_EQ(true, dynamic_cast<JitteredExponentialBackOffStrategy*>(strategy.get())
+                          ->isOverTimeLimit(test_max_interval_ms + 1));
+    }
+
+    // only set base_interval
+    {
+      envoy::config::core::v3::GrpcService grpc_service;
+
+      uint64_t test_base_interval_ms = 5000;
+
+      grpc_service.mutable_retry_policy()
+          ->mutable_retry_back_off()
+          ->mutable_base_interval()
+          ->set_seconds(test_base_interval_ms / 1000);
+
+      BackOffStrategyPtr strategy =
+          Utility::prepareBackoffStrategy<envoy::config::core::v3::GrpcService>(grpc_service,
+                                                                                random);
+
+      // max_interval should be less than or equal test_base_interval * 10
+      EXPECT_EQ(false, dynamic_cast<JitteredExponentialBackOffStrategy*>(strategy.get())
+                           ->isOverTimeLimit(test_base_interval_ms * 10));
+      EXPECT_EQ(true, dynamic_cast<JitteredExponentialBackOffStrategy*>(strategy.get())
+                          ->isOverTimeLimit(test_base_interval_ms * 10 + 1));
+    }
+
+    // set max_interval > base_interval
```

Comment:
Do you mean max_interval < base_interval here?

Reply:
will fix ,thanks a lot!